test_conformance/common/vulkan_wrapper/vulkan_wrapper.cpp

```diff
@@ -130,6 +152,25 @@ VulkanInstance::VulkanInstance(): m_vkInstance(VK_NULL_HANDLE)
     VK_GET_NULL_INSTANCE_PROC_ADDR(vkCreateInstance);
 #undef VK_GET_NULL_INSTANCE_PROC_ADDR
 
+    if (m_useValidationLayers)
+    {
+        uint32_t layerCount = 0;
+        vkEnumerateInstanceLayerProperties(&layerCount, nullptr);
+
+        std::vector<VkLayerProperties> layers(layerCount);
+        vkEnumerateInstanceLayerProperties(&layerCount, layers.data());
+
+        bool found = false;
+        for (const char *layerName : m_validationLayers)
+            for (const auto &layerProps : layers)
+                if (strcmp(layerName, layerProps.layerName) == 0)
+                {
+                    found = true;
+                    break;
+                }
+        m_useValidationLayers = found;
```

Comment:
This will enable the validation layers if _any_ of the requested validation layers are found, even if _some_ of the requested validation layers are not found.  Is this the behavior that we want, or do we only want to enable validation layers if _all_ of the requested validation layers are found?

Actually, since it looks like we use all validation layers when we build our VkInstanceCreateInfo, we should definitely be checking that all validation layers are found, or we need to ensure we only enable the validation layers that were found.

Reply:
Corrected